test/e2e/reloadinterval_test.go

```diff
@@ -0,0 +1,133 @@
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestReloadInterval(t *testing.T) {
+	t.Parallel()
+
+	// values that do not satisfy the regex validation will generate this error
+	const expectedErr string = `spec.tuningOptions.reloadInterval in body should match '^(0|([0-9]+(\.[0-9]+)?(s|m))+)$'`
+
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "reloadinterval-test"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	ic := newPrivateController(icName, domain)
+
+	if err := kclient.Create(context.TODO(), ic); err != nil {
+		t.Fatalf("failed to create ingresscontroller %s: %v", icName, err)
+	}
+
+	defer assertIngressControllerDeleted(t, kclient, ic)
+	t.Logf("waiting for ingresscontroller %v", icName)
+	if err := waitForIngressControllerCondition(t, kclient, 5*time.Minute, icName, availableConditionsForPrivateIngressController...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	ic, err := getIngressController(t, kclient, icName, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	deployment := &appsv1.Deployment{}
+	if err := kclient.Get(context.TODO(), controller.RouterDeploymentName(ic), deployment); err != nil {
+		t.Fatalf("failed to get ingresscontroller router deployment: %v", err)
+	}
+
+	for _, testCase := range []struct {
+		description    string
+		reloadInterval metav1.Duration
+		expectedEnvVar string
+		expectSuccess  bool
+	}{
+		// set ReloadInterval to a value that does not pass the regex validation
+		// expect validation error
+		{"set reloadInterval 0.5s", metav1.Duration{500 * time.Millisecond}, "5s", false},
+		{"set reloadInterval 1s", metav1.Duration{-1 * time.Second}, "5s", false},
+
+		// the following values all pass the regex validation
+		//
+		// set Spec.TuningOptions.ReloadInterval to a nondefault value within allowed bounds
+		// expect no error
+		{"set reloadInterval 100s", metav1.Duration{100 * time.Second}, "100s", true},
+		{"set reloadInterval 1m", metav1.Duration{1 * time.Minute}, "1m", true},
+
+		// set ReloadInterval to 0, which should give default of 5s
+		// expect no error
+		{"set reloadInterval default", metav1.Duration{0 * time.Second}, "5s", true},
+
+		// set ReloadInterval to a value outside of allowed bounds
+		// expect no error
+		{"set reloadInterval 130s", metav1.Duration{130 * time.Second}, "2m", true},
+	} {
+		t.Log(testCase.description)
+		// cases with values that pass the regex validation
+		if testCase.expectSuccess {
+			if err := setReloadIntervalValidValue(t, kclient, 1*time.Minute, testCase.reloadInterval, icName); err != nil {
+				t.Fatalf("failed to update ingresscontroller with reloadInterval=%v: %v", testCase.reloadInterval, err)
```

Comment:
It would be good if this continued on to do the other tests, so if you use `Errorf` instead of `Fatalf` you can continue instead of exiting.  
```suggestion
				t.Errorf("failed to update ingresscontroller with reloadInterval=%v: %v", testCase.reloadInterval, err)
```